src/view/components/microbit/MicrobitImage.tsx

```diff
@@ -5,6 +5,73 @@ import * as React from "react";
 import "../../styles/Microbit.css";
 import { MICROBIT_SVG } from "./Microbit_svg";
 
-export const MicrobitImage: React.FC = () => {
-    return MICROBIT_SVG;
+interface EventTriggers {
+    onMouseUp: (button: HTMLElement, event: Event) => void;
+    onMouseDown: (button: HTMLElement, event: Event) => void;
+    onMouseLeave: (button: HTMLElement, event: Event) => void;
+}
+interface IProps {
+    eventTriggers: EventTriggers;
+    leds: number[][];
+}
+interface IState {
+    microbitImageReference: React.RefObject<SVGSVGElement>;
+}
+
+// Displays the SVG and call necessary svg modification.
+export class MicrobitImage extends React.Component<IProps, IState> {
+    constructor(props: IProps) {
+        super(props);
+        this.state = { microbitImageReference: React.createRef() };
+    }
+    componentDidMount() {
+        const svgElement = this.state.microbitImageReference.current;
+        if (svgElement) {
+            setupAllButtons(this.props.eventTriggers);
+            updateAllLeds(this.props.leds);
+        }
+    }
+    componentDidUpdate() {
+        updateAllLeds(this.props.leds);
+    }
+    render() {
+        return MICROBIT_SVG(this.state.microbitImageReference);
+    }
+}
+const setupButton = (
+    buttonElement: HTMLElement,
+    eventTriggers: EventTriggers
+) => {
+    buttonElement.onmousedown = e => {
+        eventTriggers.onMouseDown(buttonElement, e);
+    };
+    buttonElement.onmouseup = e => {
+        eventTriggers.onMouseUp(buttonElement, e);
+    };
+
+    buttonElement.onmouseleave = e => {
+        eventTriggers.onMouseLeave(buttonElement, e);
+    };
+};
+const setupAllButtons = (eventTriggers: EventTriggers) => {
+    const buttonsId = ["BTN_A_OUTER", "BTN_B_OUTER", "BTN_AB_OUTER"];
+    buttonsId.forEach(buttonId => {
+        const buttonElement = window.document.getElementById(buttonId);

```

Comment:
I assume we'd probably want to use refs here as well, as Sagar mentioned.

Reply:
Yes, I've actually made this change on the other opened PR